File: backend/routes/articles.js
const express = require("express");
const router = express.Router();
const articleController = require("../controllers/articleController");
const requireAuth = require("../middleware/requireAuth");

router.get("/", articleController.getAll);
router.get("/by-author", articleController.getByAuthor);
router.get("/by-author/:author", articleController.getByAuthor);
router.post("/create", requireAuth, articleController.create);
router.get("/:id", articleController.getById);
router.delete("/:id", requireAuth, articleController.deleteById);

module.exports = router;

File: backend/routes/articles.py
from flask import Blueprint
from controllers.articleController import *

articles = Blueprint("articles", "backend")  # FIXME: don't hardcode backend

articles.add_url_rule("/", "get_all_articles", get_all_articles, methods=["GET"])
articles.add_url_rule("/<id>", "get_article", get_article, methods=["GET"])
articles.add_url_rule("/", "post_article", post_article, methods=["POST"])
articles.add_url_rule("/<id>", "delete_article", delete_article, methods=["DELETE"])
articles.add_url_rule(
    "/by-author/<author>",
    "get_article_by_author",
    get_article_by_author,
    methods=["GET"],
)

# Not implemented yet in the frontend
# articles.add_url_rule(put_article, "/<id>", methods=["PUT"])
